hedera-node/src/main/java/com/hedera/services/store/tokens/unique/UniqueTokenStore.java

```diff
@@ -144,6 +149,71 @@ public MerkleUniqueToken get(final NftID id) {
 		return uniqueTokenSupplier.get().get(fromNftID(id));
 	}
 
+	@Override
+	public ResponseCodeEnum adjustBalance(AccountID senderAId, AccountID receiverAId, TokenID tId, long serialNumber) {
+		var validity = sanityChecked(senderAId, receiverAId, tId, token -> tryAdjustment(senderAId, receiverAId, tId, serialNumber));
+		if (validity == OK) {
+			var nftId = NftID.newBuilder().setTokenID(tId).setSerialNumber(serialNumber).build();
+			if (nftExists(nftId)) {
+				get(nftId).setOwner(EntityId.fromGrpcAccountId(receiverAId));
+
+				adjustOwnedNfts(senderAId, false);
+				adjustOwnedNfts(receiverAId, true);
+			} else {
+				validity = INVALID_NFT_ID;
+			}
+		}
+
+		return validity;
+	}
+
+	public void adjustOwnedNfts(AccountID aId, Boolean isAdded) {
+		var merkleAccount = hederaLedger.get(aId);
+		long nftsOwned = merkleAccount.getNftsOwned();
+		if (Boolean.TRUE.equals(isAdded)) {
+			merkleAccount.setNftsOwned(nftsOwned + 1);
+		} else {
+			merkleAccount.setNftsOwned(nftsOwned - 1);
+		}
+	}
+
+	private ResponseCodeEnum tryNftAdjustment(AccountID aId, TokenID tId, long serialNumber) {
```

Comment:
This is invalid. We can mint `n` number of NFTs (having `n` serialNumbers`). The argument that you are providing is passing only the last `serialNumber` minted and we must have entries in the record file for all serialNumbers minted

Reply:
Good catch. It's fixed! :)